src/org/labkey/test/components/ui/grids/QueryGrid.java

```diff
@@ -435,7 +435,7 @@ public void saveView(String viewName, boolean makeAvailable)
                 dialog.setMakeAvailable(false);
         }
 
-        dialog.saveView();
+        doAndWaitForUpdate(()-> dialog.saveView());
```

Comment:
I don't think this is the right fix.
I think the check for a stale element, or preventing one, should be done in the getViewName method.

Reply:
I disagree. The method that causes the element to go stale should be responsible for wait for the operation to complete.

Reply:
The problem is that there is another save view methods that only returns the dialog. This fix would not address any of those tests that use that method.
Why not change the find in getEditAlertText to be a refindWhenNeeded?

Reply:
Looks like `SaveViewDialog` has access to the grid. Could we move the call to `doAndWaitForUpdate` into the `saveView` method?

Reply:
It's worth a try!